admin2: Add GUI panel for mute

omar-o22 opened the pull request:
Summary
-
- This PR adds a proper mute system with a new GUI and fixes #108.  
- Previously, mutes were just a toggle and expired on reconnect.  
- Now, admins can set a duration, and the mute duration only counts while the player is connected (it pauses if the player disconnects).

Preview:
-
<img width="747" height="600" alt="image" src="https://github.com/user-attachments/assets/95d3bc56-d266-4e82-9387-827a8e243441" />

Edited
-
I did a mention for issue 106 by mistake

omar-o22 committed:
Add GUI panel for mute

omar-o22 committed:
Fix lint warning

ArranTuna reviewed (changes requested):
If you close admin panel while the mute GUI is open, it doesn't close, and then after that I clicked close (with console) and GUI won't close.

Fernando-A-Rocha commented:
I agree mute duration should be minutes by default, but also allow other units of measurement for custom needs (seconds, hours..)

ArranTuna commented:
You're right, what you have to do is, press mute twice, then close the panel and you're stuck with a second panel. So basically when you click mute, don't make another GUI if one already exists.

jlillis reviewed (dismissed):
I don't think this quite resolves #108, as it mutes are not able to persist if the server or resource restarts. There will also need to be a new tab to view, modify, and delete existing mutes.

omar-o22 committed:
Merge branch 'master' into feature/add-GUI-mute

ArranTuna commented:
> > You're right, what you have to do is, press mute twice, then close the panel and you're stuck with a second panel. So basically when you click mute, don't make another GUI if one already exists.
> 
> I don't know where to save the data because admin2 saves some data in conf/***.xml and some in admin.db too So should I save data in admin.db or create a new xml file?

Use this existing connection to admin.db:

connection = dbConnect("sqlite", "admin.db"),

And use the dbExec and dbQuery functions and use dbQuery call back function feature.

omar-o22 committed:
Fixes

- Added a new tab 'mutes'
- Created a new event for mute instead of "aPlayer"
- Saves the data in database

omar-o22 committed:
update

omar-o22 committed:
I pushed the last commit by mistake

omar-o22 committed:
Merge remote-tracking branch 'origin/master' into feature/add-GUI-mute

omar-o22 committed:
Fix lint warnings

ArranTuna commented on [admin]/admin2/server/admin_mute.lua:
```diff
@@ -0,0 +1,273 @@
+    local query_text = "SELECT serial FROM mutes WHERE serial = ?"
+    if ( isRestarted ) then
+        local query = dbQuery( db.connection, query_text, serial )
+        result = dbPoll( query, -1 )
```
You should be using the call back feature.

ArranTuna commented:
I added a mute on myself and don't see it in mutes tab and refresh button is disabled?

omar-o22 committed:
Update ACL.xml

omar-o22 committed:
Merge branch 'master' into feature/add-GUI-mute

Lpsd reviewed (approved):
Looks good to me.

As mentioned in Discord, may be nice to have an option/checkbox for muting player voice too, if that's possible.

omar-o22 committed:
Merge branch 'master' into feature/add-GUI-mute

omar-o22 committed:
Merge branch 'master' into feature/add-GUI-mute

ArranTuna committed:
Merge branch 'master' into feature/add-GUI-mute

ArranTuna committed:
Merge branch 'master' into feature/add-GUI-mute

ArranTuna reviewed (approved):
I will just make the change myself about using the db call back
